src/charm.py

```diff
@@ -94,12 +96,34 @@ def __init__(self, *args):
             plural="ipaddresspools",
         )
 
+        self.L2Advertisement = create_namespaced_resource(
+            group="metallb.io",
+            version="v1beta1",
+            kind="L2Advertisement",
+            plural="l2advertisements",
+        )
+
```

Comment:
L2Advertisements are a requirement of metallb in order for it to answer ARP requests for the services being advertised through the load-balancer.